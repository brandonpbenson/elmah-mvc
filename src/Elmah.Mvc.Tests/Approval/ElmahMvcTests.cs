using System.Text.RegularExpressions;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Elmah.Mvc.Tests.Approval
{
    [UseReporter(typeof(DiffReporter))]
    public class ElmahMvcTests
    {
        private readonly string ElmahMvcAppUrl = "http://localhost:49800";

        [Test]
        public void should_open_elmah_page(string path)
        {
            // do
            var url = ElmahMvcAppUrl + path;

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_stylesheets()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/stylesheet";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/rss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_digest_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/digestrss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_details()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_not_found_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_json_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/json?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_xml_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/xml?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        private string GetContent(string url)
        {
            return HackServertTime(ApprovalTests.Asp.AspApprovals.GetUrlContents(url));
        }

        private string HackServertTime(string content)
        {
            return new Regex("Server time is \\d+:\\d+:\\d+").Replace(content, string.Empty);
        }
    }
}
