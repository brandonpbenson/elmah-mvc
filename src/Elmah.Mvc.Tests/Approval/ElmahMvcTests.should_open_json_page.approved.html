{ "application": "/", "host": "SEEKEY-NOTE", "type": "Elmah.ApplicationException", "message": "Missing error identifier specification.", "source": "Elmah", "detail": "Elmah.ApplicationException: Missing error identifier specification.\r\n   at Elmah.ErrorXmlHandler.ProcessRequest(HttpContext context)\r\n   at Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahResult.ExecuteResult(ControllerContext context) in D:\\Development\\Projects\\elmah.mvc.controller\\src\\Elmah.Mvc.Web\\Areas\\Admin\\Controllers\\ElmahController.cs:line 93\r\n   at System.Web.Mvc.ControllerActionInvoker.InvokeActionResult(ControllerContext controllerContext, ActionResult actionResult)\r\n   at System.Web.Mvc.ControllerActionInvoker.<>c__DisplayClass1c.<InvokeActionResultWithFilters>b__19()\r\n   at System.Web.Mvc.ControllerActionInvoker.InvokeActionResultFilter(IResultFilter filter, ResultExecutingContext preContext, Func`1 continuation)\r\n   at System.Web.Mvc.ControllerActionInvoker.<>c__DisplayClass1c.<>c__DisplayClass1e.<InvokeActionResultWithFilters>b__1b()\r\n   at System.Web.Mvc.ControllerActionInvoker.InvokeActionResultWithFilters(ControllerContext controllerContext, IList`1 filters, ActionResult actionResult)\r\n   at System.Web.Mvc.ControllerActionInvoker.InvokeAction(ControllerContext controllerContext, String actionName)\r\n   at System.Web.Mvc.Controller.ExecuteCore()\r\n   at System.Web.Mvc.ControllerBase.Execute(RequestContext requestContext)\r\n   at System.Web.Mvc.ControllerBase.System.Web.Mvc.IController.Execute(RequestContext requestContext)\r\n   at System.Web.Mvc.MvcHandler.<>c__DisplayClass6.<>c__DisplayClassb.<BeginProcessRequest>b__5()\r\n   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass1.<MakeVoidDelegate>b__0()\r\n   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)\r\n   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()\r\n   at System.Web.Mvc.MvcHandler.<>c__DisplayClasse.<EndProcessRequest>b__d()\r\n   at System.Web.Mvc.SecurityUtil.<GetCallInAppTrustThunk>b__0(Action f)\r\n   at System.Web.Mvc.SecurityUtil.ProcessInApplicationTrust(Action action)\r\n   at System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult)\r\n   at System.Web.Mvc.MvcHandler.System.Web.IHttpAsyncHandler.EndProcessRequest(IAsyncResult result)\r\n   at System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()\r\n   at System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously)", "time": "2012-06-08T15:38:57.5852713Z", "serverVariables": { "ALL_HTTP": "HTTP_CONNECTION:keep-alive\r\nHTTP_ACCEPT:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nHTTP_ACCEPT_CHARSET:ISO-8859-1,utf-8;q=0.7,*;q=0.3\r\nHTTP_ACCEPT_ENCODING:gzip,deflate,sdch\r\nHTTP_ACCEPT_LANGUAGE:uk-UA,uk;q=0.8,ru;q=0.6,en-US;q=0.4,en;q=0.2,sv;q=0.2\r\nHTTP_COOKIE:expected_tab=twitter; welcome_info_name=Alexander%20Beletsky\r\nHTTP_HOST:localhost:49800\r\nHTTP_USER_AGENT:Mozilla/5.0 (Windows NT 6.1) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.52 Safari/536.5\r\n", "ALL_RAW": "Connection: keep-alive\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.3\r\nAccept-Encoding: gzip,deflate,sdch\r\nAccept-Language: uk-UA,uk;q=0.8,ru;q=0.6,en-US;q=0.4,en;q=0.2,sv;q=0.2\r\nCookie: expected_tab=twitter; welcome_info_name=Alexander%20Beletsky\r\nHost: localhost:49800\r\nUser-Agent: Mozilla/5.0 (Windows NT 6.1) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.52 Safari/536.5\r\n", "APPL_PHYSICAL_PATH": "D:\\Development\\Projects\\elmah.mvc.controller\\src\\Elmah.Mvc.Web\\", "AUTH_PASSWORD": "*****", "LOGON_USER": "seekey-note\\alexander.beletsky", "CONTENT_LENGTH": "0", "LOCAL_ADDR": "::1", "PATH_INFO": "/admin/elmah/xml", "PATH_TRANSLATED": "D:\\Development\\Projects\\elmah.mvc.controller\\src\\Elmah.Mvc.Web\\admin\\elmah", "REMOTE_ADDR": "::1", "REMOTE_HOST": "::1", "REQUEST_METHOD": "GET", "SCRIPT_NAME": "/admin/elmah", "SERVER_NAME": "localhost", "SERVER_PORT": "49800", "SERVER_PORT_SECURE": "0", "SERVER_PROTOCOL": "HTTP/1.1", "URL": "/admin/elmah", "HTTP_CONNECTION": "keep-alive", "HTTP_ACCEPT": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "HTTP_ACCEPT_CHARSET": "ISO-8859-1,utf-8;q=0.7,*;q=0.3", "HTTP_ACCEPT_ENCODING": "gzip,deflate,sdch", "HTTP_ACCEPT_LANGUAGE": "uk-UA,uk;q=0.8,ru;q=0.6,en-US;q=0.4,en;q=0.2,sv;q=0.2", "HTTP_COOKIE": "expected_tab=twitter; welcome_info_name=Alexander%20Beletsky", "HTTP_HOST": "localhost:49800", "HTTP_USER_AGENT": "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.52 Safari/536.5" }, "cookies": { "expected_tab": "twitter", "welcome_info_name": "Alexander%20Beletsky" } }